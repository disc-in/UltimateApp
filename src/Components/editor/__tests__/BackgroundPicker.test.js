import React from 'react';
import { render, waitFor } from '@testing-library/react-native';

import BackgroundPicker from '../BackgroundPicker';

describe('<BackgroundPicker />', () => {
  afterEach(() => jest.clearAllMocks());

  const onBackgroundChange = jest.fn();
  const selectedBackground = 'endzone';

  it('renders correctly', async () => {
    const { toJSON } = await waitFor(() =>
      render(<BackgroundPicker onBackgroundChange={onBackgroundChange} selectedBackground={selectedBackground} />),
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
