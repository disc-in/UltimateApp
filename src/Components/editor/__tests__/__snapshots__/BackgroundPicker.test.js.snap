// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BackgroundPicker /> renders correctly 1`] = `
<View
  style={
    Object {
      "height": 80,
      "width": "45%",
    }
  }
>
  <RNCPicker
    items={
      Array [
        Object {
          "label": "Rectangle",
          "testID": undefined,
          "textColor": undefined,
          "value": "rectangle",
        },
        Object {
          "label": "Endzone",
          "testID": undefined,
          "textColor": undefined,
          "value": "endzone",
        },
        Object {
          "label": "3/4 field",
          "testID": undefined,
          "textColor": undefined,
          "value": "three quarters field",
        },
        Object {
          "label": "Empty",
          "testID": undefined,
          "textColor": undefined,
          "value": "empty",
        },
      ]
    }
    numberOfLines={1}
    onChange={[Function]}
    selectedIndex={1}
    style={
      Array [
        Object {
          "height": 216,
        },
        Object {
          "fontSize": 14,
          "height": 80,
        },
      ]
    }
  />
</View>
`;
